### DISCUSSION TITLE
sorting is still slow

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#5472] [Q&A] [answered]
here is a reproducible example, is there a way to improve the performance of this pipeline, currently it take 10 minutes on Colab

https://colab.research.google.com/drive/1pfAPpIG7jpvGB_aHj-PXX66vRaRT0xlj?usp=sharing

```
main code here
import pyarrow.dataset as ds
import duckdb 
con = duckdb.connect()
result =con.execute(
'''
SET memory_limit='10GB';
SET temp_directory='./';
select *,extract ( year from l_shipdate) as year  from 'lineitem.parquet' order by l_shipdate
''').fetch_record_batch()
ds.write_dataset(result,"./lineitem/", format="parquet" , partitioning=['year'],partitioning_flavor="hive",max_rows_per_group=120000,existing_data_behavior="overwrite_or_ignore")
```

**Comment by hannes** [comment, marked as answer]
Are you sure it's the duckdb sorting that is slow here and not ds.write_dataset?

**Reply by djouallah** [reply]
using the native DuckDB copy, gave more or less the same performance 12 minutes

```
%%time
import pyarrow.dataset as ds
import duckdb 
con = duckdb.connect()
result =con.execute(
'''
SET memory_limit='10GB';
SET temp_directory='./';
COPY  (select *,extract ( year from l_shipdate) as year  from 'lineitem.parquet' order by l_shipdate) to 'lineitem_sorted.parquet' (FORMAT 'PARQUET', CODEC 'ZSTD')
''').df()
result
```

**Reply by hannes** [reply]
Okay, but you are still not measuring the sort code most likely but the parquet writer performance instead.

**Reply by lnkuiper** [reply]
I assume this is SF10? Your query takes around 1 minute on my laptop, also with a 10GB memory limit, but does require spilling data to disk, as the data does not fit fully in memory. You are mostly measuring the disk speed on colab, rather than the DuckDB's sort speed.

Our sort works best when the data fits in memory. When it does not, the disk speed heavily impacts performance.
